patches/comfyui_3d_pack.diff

```diff
@@ -1,3 +1,218 @@
+diff --git a/Gen_3D_Modules/Unique3D/custum_3d_diffusion/custum_pipeline/unifield_pipeline_img2img.py b/Gen_3D_Modules/Unique3D/custum_3d_diffusion/custum_pipeline/unifield_pipeline_img2img.py
+index 0809ab0..49d6c22 100644
+--- a/Gen_3D_Modules/Unique3D/custum_3d_diffusion/custum_pipeline/unifield_pipeline_img2img.py
++++ b/Gen_3D_Modules/Unique3D/custum_3d_diffusion/custum_pipeline/unifield_pipeline_img2img.py
+@@ -13,10 +13,11 @@
+ # limitations under the License.
+ # modified by Wuvin
+ 
+ from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+ 
++import comfy.utils
+ import numpy as np
++import PIL
+ import torch
+ 
+ from diffusers import AutoencoderKL, UNet2DConditionModel, StableDiffusionImageVariationPipeline
+@@ -96,10 +97,97 @@ class StableDiffusionImageCustomPipeline(
+ 
+         return image_embeddings
+ 
++    # adapted from https://github.com/huggingface/diffusers/blob/v0.12.0/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L471
++    def get_timesteps(self, num_inference_steps, new_steps, denoise):
++        # get the original timestep using init_timestep
++        init_timestep = min(int(num_inference_steps * denoise),
++                            num_inference_steps)
++        if init_timestep == 0:
++            raise ValueError(f"denoise {denoise} is too small.")
++        t_start = max(new_steps - num_inference_steps, 0)
++        timesteps = self.scheduler.timesteps[t_start:]
++
++        return timesteps, new_steps - t_start
```

Comment:
if it is copied unchanged, let's just call it, but pass the current "self" even if it is a different class object -- only field used is just scheduler